src/main/java/core/basesyntax/service/Handle.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service;
+
+import core.basesyntax.products.Fruit;
+import core.basesyntax.products.FruitDto;
+import java.util.HashMap;
+import java.util.List;
+
+public class Handle {
+    private static final HashMap<String, ServiceAble> handleCheck = new HashMap<>();
+
+    static {
+        handleCheck.put("s", new SupplyService());
+        handleCheck.put("b", new BuyService());
+        handleCheck.put("r", new ReturnService());
+    }
+
+    public boolean operationWithProduct(List<FruitDto> fruitDtoList) {
+        for (FruitDto dto : fruitDtoList) {
+            if (handleCheck.containsKey(dto.getOperation()) == false) {
```

Comment:
'handleCheck.containsKey(dto.getOperation()) == false' can be simplified to '!handleCheck.containsKey(dto.getOperation())'  - IDEA

Reply:
Difficult to see the ! here, did it special for easy reading.

Reply:
We were talking about this on lessons, it is so easy to lose this small sign ! on this code so I did it special, even the IDEA told me the same it could be replaced.